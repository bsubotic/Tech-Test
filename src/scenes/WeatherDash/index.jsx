import React, { useState, useEffect } from "react";

import { getAllCities } from "services/api/cities";

import CityCard from "./Components/CityCard";

export default () => {
  const [cities, setCities] = useState(null);

  useEffect(() => {
    getAllCities().then(cities => setCities(cities));
  }, []);

  return (
    <div className="pa4 flex flex-nowrap overflow-x-auto h-100">
      {cities &&
        cities.map(cityName => <CityCard key={cityName} cityName={cityName} />)}
    </div>
  );
};
