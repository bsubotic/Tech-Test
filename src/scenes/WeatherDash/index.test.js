import React from "react";
import renderer from "react-test-renderer";

import WeatherDash from "./index";

test("render an enabled submit button", () => {
  const tree = renderer.create(<WeatherDash />).toJSON();
  expect(tree).toMatchSnapshot();
});
