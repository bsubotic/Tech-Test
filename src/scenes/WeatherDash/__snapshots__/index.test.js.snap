// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an enabled submit button 1`] = `
<div
  className="pa4 flex flex-nowrap overflow-x-auto h-100"
/>
`;
